lib/core/settle.js

```diff
@@ -11,7 +11,8 @@ var createError = require('./createError');
  */
 module.exports = function settle(resolve, reject, response) {
   var validateStatus = response.config.validateStatus;
-  if (!validateStatus || validateStatus(response.status)) {
+  var responseUrl = response.config.url;
+  if (!validateStatus || validateStatus(response.status) || (!response.status && ~responseUrl.indexOf('file:'))) {
```

Comment:
the tilde bitwise logic returns -1 when the string doesn't match and 0 when it matches, I believe it will always be false? unless I'm missing something
```javascript
> myString = "test123"
"test123"
> ~myString.indexOf("test");
-1
> ~myString.indexOf("notpresent");
0
```

You can just do the normal `.indexOf('file:') >= 0` or something similar

Reply:
hi, `-1` if true rather than false... 
```
var i = -1;
if (i) {
    console.log(i, " is true");
} else {
    console.log(i, " is false");
}
```
it will ouput: `-1 ' is true'`

Reply:
🤦‍♂️ right